### ISSUE TITLE
PEFT/LoRA resume silently restores no adapter weights

### ISSUE THREAD

**Issue by evgri243** [JetBrains-Research/opaque#1041]
## Problem

`_load_model_weights` ends with:

```python
# packages/opaque-transformers/src/opaque/api/transformers/trainer/_dp_trainer.py:5479-5481
if not mutated:
    strict = not self._is_peft
    self._model.load_state_dict(new_state, strict=strict)
```

`_load_best_model` repeats those same three lines verbatim (`:4910-4912`). For a PEFT-wrapped model `self._is_peft` is `True` (`:467`), so the load runs with `strict=False` — which is fatal here, because **the two key spaces do not overlap at all**:

- `_save_model_artifacts` (`:5244-5247`) checkpoints through `self._model.save_pretrained(...)`, and PEFT's `save_pretrained` strips the adapter name: `base_model.model...lora_A.weight`.
- The live module's `state_dict()` keys carry it: `base_model.model...lora_A.default.weight`.

Every saved key is therefore "unexpected" and every model key "missing". The returned `_IncompatibleKeys` is discarded and nothing is logged. Measured on the literal call at `:5481`: 4 unexpected / 33 missing keys, nothing restored, no exception.

There is no second restore path — `_apply_runtime_state` (`:5548-5573`) rewrites `ctx.trainable_params` only for `ScheduleFreeState`, so the trainable parameters come from the unmutated module.

Hugging Face's `Trainer`, whose behaviour DPTrainer claims parity with, uses `model.load_adapter(resume_from_checkpoint, active_adapter, is_trainable=True)` for exactly this case (`transformers/trainer.py:3440` on resume, `:3502` for best-model load). A grep for `set_peft_model_state_dict`, `get_peft_model_state_dict` or `load_adapter` across `packages/` returns nothing.

## Impact

Any PEFT/LoRA run — the documented DPTrainer fine-tuning shape — resumed with `resume_from_checkpoint`. Reproduced end to end on the real trainer (peft 0.20.0, transformers 5.15.1; tiny GPT-2 + LoRA r=4 on `c_attn`, 4 steps, `save_steps=2`, `use_cpu`). Instrumenting the resume prints:

```
_is_peft: True
weights changed by load?: False
post-load == checkpoint?: False
post-load == fresh-init?: True
```

The run then completes to `global_step=4` reporting `privacy_epsilon=0.8132` with no warning. With dropout disabled so the trajectory is otherwise deterministic, the continuous run logs `grad_norm` 0.836132824 / 0.929508567 at steps 3-4 while the resumed run logs 1.277820706 / 1.491086841 on the same batches and the same reported epsilon; final `|Δw|` over the LoRA tensors is 0.1716.

**The user gets a plausible-looking completed DP run whose model has thrown away all prior learning — and in a DP setting that loss is not recoverable by re-running, because the dataset's privacy budget for those steps has already been spent.**

The identical three lines at `:4910-4912` make `load_best_model_at_end=True` a no-op for PEFT too: reproduced with the best checkpoint recorded as checkpoint-2, the final in-memory adapter equals checkpoint-4 (the last), not the best — the outcome the flag's own docstring at `:4877-4880` says it exists to prevent.

Every existing resume test asserts only `global_step` and `privacy_epsilon`, never restored weights (`tests/validation/test_dp_guarantees.py:347-349, :567-572`). `docs/user-guide/huggingface/dptrainer.md:79-88` documents resume with no caveat.

## Acceptance criteria

- Both `_load_model_weights` and `_load_best_model` branch on `self._is_peft` and call `peft.set_peft_model_state_dict(self._model, new_state, adapter_name=self._model.active_adapter)`, or `self._model.load_adapter(ckpt_dir, active_adapter, is_trainable=True)` matching upstream.
- **Independently of the PEFT branch:** capture the `_IncompatibleKeys` result at both sites and raise when *no* key matched. That single assertion converts this whole failure class from silent to loud for any future key-space skew, and is the cheapest high-leverage change in the Granite set.
- A regression test checkpoints a LoRA run, resumes in a fresh process, and asserts the live adapter tensors equal the checkpoint's rather than the fresh init.
- A `load_best_model_at_end=True` variant asserts the final adapter equals the *best* checkpoint, not the last.
- The existing resume tests gain a weight assertion, so "resume works" can no longer mean only "the step counter advanced".

## Notes

`affects_epsilon` is false — the reported epsilon correctly prices the mechanism that executed. What is destroyed is the *product* of that spend. Reaching it needs two explicit opt-ins (a PEFT-wrapped model plus `resume_from_checkpoint` or `load_best_model_at_end`), which is the only argument for high rather than critical; verification kept critical because the failure is silent and total at the library's primary documented fine-tuning shape, and the wasted budget is unrecoverable.

Fix together with the tied-embedding defect at the same two call sites (OPQ-448) — same functions, same shared `_IncompatibleKeys` remedy.

Audit source: OPQ-445.
